daemon/cluster/provider/cluster.go

```diff
@@ -0,0 +1,36 @@
+package provider
```

Comment:
Not without understanding the goal more.

In general, it would be good to correctly use the term provider, so it has proper meaning within the architecture. In general, I would just put this in a cluster package or push the definition to the consumer.